virttest.qemu_qtree: Update guess_type() for QtreeDev

spcui opened the pull request:
Make the get method can get the correct QtreeDisk device
on different platform.

ID: 1211151 1231011

CongSmile commented:
Hi @spcui 

The request is that virtio-blk-pci info in qtree is different in RHEL.6 and RHEL.7, we need update the code to the root cause I think.
I don't think that disabling the virtio-blk-pci can solve the problem.

Please be free to correct me if I'm misunderstanding.

Thanks.

spcui commented:
Hi @CongSmile 

For rhel7 host, self.qtree['type'] == 'virtio-blk-device', and return disk, 
for rhel6 host, self.qtree['type'] == 'virtio-blk-pci', and elif branch can return the correct type.

CongSmile commented:
Thx for the explanation at first.

For the elif branch, you don't judge "virtio-blk-pci", right?
 which means that no matter qtree['type'] equals to "virtio-blk-pci", it will return the QtreeDisk.

But actually, it qtree['type'] != "virtio-blk-pci", it should return QtreeDev, right?

Please be free to correct me if I'm misunderstanding.

Thanks.

Commit:
virttest.qemu_qtree: Update guess_type() for QtreeDev

Make the get method can get the correct QtreeDisk device
on different platform.

Signed-off-by: Shuping Cui <scui@redhat.com>

spcui commented:
HI @CongSmile updated, could you help review again?

CongSmile commented:
Acked-by: Cong Li <coli@redhat.com>

xutian commented:
@spcui 

for guess_type in Qdev,  it just get dev objects in qtree, for guess_type in qdisk, it should be overwrite in Qdisk not in Qdev.
so Nack this PR.

Thanks,
Xu

spcui commented:
@xutian @CongSmile updated, could you review again?

CongSmile commented:
Hi @spcui 

CI check failed, could you fix it at first?

Thx.

vipmike007 commented on virttest/qemu_qtree.py:
```diff
@@ -471,12 +479,14 @@ def check_guests_proc_scsi(self, info):
-            logging.error('Disk %s is in qtree but not in /proc/scsi/scsi.',
-                          disk)
+            error_msg = 'Disk %s is in qtree but not in /proc/scsi/scsi' % disk
+            logging.error(error_msg)
```
Does this function work for windows?
If yes looks the error_msg is not correct.

vipmike007 commented on virttest/qemu_qtree.py:
```diff
@@ -471,12 +479,14 @@ def check_guests_proc_scsi(self, info):
-            logging.error('Disk %s is in /proc/scsi/scsi but not in qtree.',
-                          disk)
+            error_msg = 'Disk %s is in /proc/scsi/scsi but not in qtree' % disk
+            logging.error(error_msg)
```
same as above

vipmike007 commented:
inspekt style
/home/travis/build/autotest/virt-test/virttest/qemu_qtree.py:364:5: E125 20
PEP8 check fail: /home/travis/build/autotest/virt-test/virttest/qemu_qtree.py
Trying to fix errors with autopep8
PEP8 compliance FAIL
The command "inspekt style" exited with 1.

ypu commented on virttest/qemu_qtree.py:
```diff
@@ -163,15 +163,7 @@ def add_child(self, child):
-            return QtreeDisk
-        else:
-            return QtreeDev
+        return QtreeDev
```
Your update will make the guess_type meaningless.

Can you separated your pullreq to two? one is for the debug log and another is for guess_type. We can make the debug log one in tree first.

lmr commented:
Hi @spcui. As you might know, we're moving virt-test to avocado-vt, so I please ask you to consider re-implementing this functionality in the later. The new repo is:

https://github.com/avocado-framework/avocado-vt

Please keep in mind the following guidelines when creating the new PR:

1) Most of the time you can pick the patches of your original PR and merge them on a new branch with minimal changes.
3) Don't use autotest APIs, since we're moving away from using them. Please review the code and change things like `utils.system` -> `process.system`, `utils.run` -> `process.run`, so on and so forth.

Thanks, and I'm sorry for any inconveniences we might have caused your team.
